includes/admin/marketplace-suggestions/class-wc-marketplace-suggestions.php

```diff
@@ -0,0 +1,167 @@
+		if ( ! self::allow_suggestions() ) {
+			return;
+		}
+
+		// Register ajax api handlers.
+		add_action( 'wp_ajax_woocommerce_marketplace_suggestions', array( __CLASS__, 'get_suggestion_json_data_handler' ) );
+		add_action( 'wp_ajax_woocommerce_add_dismissed_marketplace_suggestion', array( __CLASS__, 'post_add_dismissed_suggestion_handler' ) );
+
+		// Register hooks for rendering suggestions container markup.
+		add_action( 'wc_marketplace_suggestions_products_empty_state', array( __CLASS__, 'render_products_list_empty_state' ) );
+	}
+
+	/**
+	 * Suggestion data GET handler.
+	 */
+	public static function get_suggestion_json_data_handler() {
+		$suggestion_data = self::get_suggestions_api_data();
+		wp_send_json_success( $suggestion_data );
+	}
+
+	/**
+	 * Return an array of suggestions the user has dismissed.
+	 */
+	public static function get_dismissed_suggestions() {
+		$dismissed_suggestions = array();
+
+		$dismissed_suggestions_data = get_user_meta( get_current_user_id(), 'wc_marketplace_suggestions_dismissed_suggestions', true );
+		if ( $dismissed_suggestions_data ) {
+			$dismissed_suggestions = $dismissed_suggestions_data;
+			if ( ! is_array( $dismissed_suggestions ) ) {
+				$dismissed_suggestions = array();
+			}
+		}
+
+		return $dismissed_suggestions;
+	}
+
+	/**
+	 * POST handler for adding a dismissed suggestion.
+	 */
+	public static function post_add_dismissed_suggestion_handler() {
+		if ( ! check_ajax_referer( 'add_dismissed_marketplace_suggestion' ) ) {
+			wp_die();
+		}
+
+		$post_data       = wp_unslash( $_POST );
+		$suggestion_slug = sanitize_text_field( $post_data['slug'] );
+		if ( ! $suggestion_slug ) {
+			wp_die();
+		}
+
+		$dismissed_suggestions = self::get_dismissed_suggestions();
+
+		if ( in_array( $suggestion_slug, $dismissed_suggestions, true ) ) {
+			wp_die();
+		}
+
+		$dismissed_suggestions[] = $suggestion_slug;
+		update_user_meta(
+			get_current_user_id(),
+			'wc_marketplace_suggestions_dismissed_suggestions',
+			$dismissed_suggestions
+		);
+
+		wp_die();
+	}
+
+	/**
+	 * Render suggestions containers in products list empty state.
+	 */
+	public static function render_products_list_empty_state() {
+		self::render_suggestions_container( 'products-list-empty-header' );
+		self::render_suggestions_container( 'products-list-empty-body' );
+		self::render_suggestions_container( 'products-list-empty-footer' );
+	}
+
+	/**
+	 * Render a suggestions container element, with the specified context.
+	 *
+	 * @param string $context Suggestion context name (rendered as a css class).
+	 */
+	public static function render_suggestions_container( $context ) {
+		include 'templates/container.php';
+	}
+
+	/**
+	 * Should suggestions be displayed?
+	 *
+	 * User can disabled all suggestions via option.
+	 * We also currently only support English suggestions.
+	 *
+	 * @return bool
+	 */
+	public static function allow_suggestions() {
+		$option = get_option( 'woocommerce_allow_marketplace_suggestions', 'yes' );
```

Comment:
Rather than an option this should be a filter because the option has no UI.

e.g. `apply_filters( 'woocommerce_allow_marketplace_suggestions',  true )`

Reply:
I agree, this should definitely be a filter right before the function returns, so a developer can easily disable it with:

```
add_filter( 'woocommerce_allow_marketplace_suggestions', '__return_false' );
```

Reply:
Good points! Addressed in 62d4b5e395893ed1ad35413a7747e8a6eecbfc12

Also added capability check so only users who can install plugins will see suggestions.

One thing I liked about `get_option` was that it could be set via CLI, but filters are probably more well known/understood.